flybywiresim/aircraft
# fix(a380/mfd): RTE RSV, AMI defaults, trip fuel, FOB & ETA in fuel and load

Fixes #9342 

## Summary of Changes

Fixes destination ETA being displayed in local time and not UTC time.
Adds flight time prediction to MFD during preflight phase.
Fixes trip fuel (non preflight) & trip time missing.
Fixes FOB value on the top of the page. 
Enables RTE RSV input fields.
Updates AMI defaults for A380 (taxi fuel & reserve max)


## Screenshots (if necessary)


![PreFlight](https://github.com/user-attachments/assets/09924989-9dd4-4057-801e-8c0052f74449)
![FuelLoad3](https://github.com/user-attachments/assets/54904694-1781-4732-abe3-a256b2f6376d)
![RTE RSV2](https://github.com/user-attachments/assets/2cd203dd-dc18-4fd5-9276-3f32a7c2b8af)
![RTE RSV](https://github.com/user-attachments/assets/7b060b0b-c2a6-4b0d-ac8d-6512b083fa3a)

## References





## Additional context



Discord username (if different from GitHub):
bruno_pt99
## Testing instructions

Start the airplane with engines off and program a flight plan.
Fill the FUEL & Load page during preflight:
1) "TIME" should be displayed on the bottom of the page above the destination airport and the estimate shows the expected flight time.
2) Trip time should be visible next to the trip fuel value
3) Enter an absolute value in the RTE RSV field. The percentage field should become dashed. Once the deleted, the percentage value appears with a value of 5% and the absolute value corresponds to that percentage of the trip fuel.
4) Change the percentage manually and verify that the absolute value is recalculated based on the entered percentage.

Perform a normal takeoff & observe the fuel & load page once in the air:
1) RTE RSV fields should show 0.0 and the inputs disabled.
2) Prediction at the bottom of the page to the destination is now in ETA rather than flight time & header shows "UTC"
3) FOB value matches what is sensed by the aircraft on the top of the page
4) Trip fuel (Expected fuel required between present time ) & trip time (remaining flight time) fields are filled with data and are in line with the FMS predictions .
5) Extra fuel field corresponds to Destination EFOB - MIN DEST FUEL.



## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
